app/assets/javascripts/initializers/initializeCommentsPage.js.erb

```diff
@@ -410,7 +410,14 @@ var KEY_CODE_K = 75;
 var ENTER_KEY_CODE = 13;
 
 function handleKeyDown(event) {
-  if (event.ctrlKey || event.metaKey) {
+  var modifier;
+  if (navigator.userAgent.indexOf('Mac OS X') != -1) {
```

Comment:
I have to ask if there's any guarantee that 'Mac OS X' would be expected in the user agent string.

Is it ever Darwin instead? Is this a 99% fix for almost everyone using common browsers and I can ignore those edge cases?

Reply:
@djuber Solid question. I should've mentioned this in the PR description: I tested in Firefox, Safari, Chrome, and Safari on my iPad (since I and many others use a keyboard with it) and each one had `Mac OS X`. Full disclosure: I was actually surprised it worked with the iPad because it's _not_ a Mac, but the user-agent string contains `like Mac OS X` (same even for the iPhone), which anywhere outside this thread I'll shamelessly pretend was completely intentional. 😄 I got the solution from [this CSS Tricks article](https://css-tricks.com/snippets/javascript/test-mac-pc-javascript/) because Chris Coyier knows more about browsers than I do.

One concern I did have related to your question was that this may not work forever because it's now called "macOS" instead of "Mac OS X", and I don't know how long the user-agent string will be backwards-compatible, so at some point this may silently fail. Using `navigator.platform` would fail more loudly, which I typically prefer, but it'd be because the editor would likely stop working and I don't imagine that's preferable. I don't know of a way to alert us if this check is no longer what we want.

Reply:
I also trust Chris Coyier more than myself 😅 

To be honest, I wonder if checking `navigator.platform` and having a loud failure would be better. I know that a failure in this area would be very visible, _but_ it wouldn't actually stop people from being able to submit comments and posts, just from being able to use keyboard shortcuts. We're likely to become aware of the issue very quickly, and it would be a relatively quick fix.

We could use our existing `currentOS()` code

https://github.com/forem/forem/blob/5832d146de4a2b21456a277b63f630153ed745c6/app/assets/javascripts/initializers/runtime.js#L45-L63

Reply:
Unfortunately [navigator.userAgentData](https://blog.uidrafter.com/engineering/userAgentData) is still experimental :( `userAgent` is the only way to go.

Reply:
if we do want to stick with `userAgent` over the existing `Runtime` helper that checks `platform`, then I think it'd be worthwhile extracting this into another utility function in `Runtime` - as mentioned in the PR description, we're going to need this elsewhere too 😅

Reply:
Totally forgot about `Runtime`!  I love the idea, if `navigator.platform` stops working, we'll have way bigger issues than the control key not working correctly in the editor, and I second having only one centralized place to get the platform. If it's possible to use `Runtime` in here, +1 for that :D

Reply:
@aitchiss Excellent point about `Runtime`. I figured we had a place to put some utility functions, but utility functions specifically for this purpose sounds great! Do you think it makes sense to hold off on it until we know we need it or go ahead and extract now?

@rhymes Ooh, TIL about `userAgentData`! I suppose not knowing about things like that is what happens when you don't touch front-end code for like 4 years. :joy: But yeah, looks like it's [Blink-only for now](https://caniuse.com/mdn-api_navigator_useragentdata).

<img width="904" alt="Screenshot of caniuse data for the navigator.userAgentData feature. The only browsers it is enabled for are Google Chrome, Microsoft Edge, and Google Chrome for Android." src="https://user-images.githubusercontent.com/108205/128530577-f31fc986-0ff5-47e3-bbea-4aa0869cb43d.png">

Reply:
> Do you think it makes sense to hold off on it until we know we need it or go ahead and extract now?

It's not a strong opinion but because it's such a specific kind of check I'd suggest extracting it into `Runtime` now. I think when we do need it later (which seems pretty definite since we have these keypress checks so many places) we're more likely to find it there than remember we have it on the comments page 😅

Reply:
Sounds good 😄 I'll go ahead and do that 👍🏼